hubstorage/resourcetype.py

```diff
@@ -26,7 +28,9 @@ def _iter_lines(self, _path, **kwargs):
 
         r = self.client.request(**kwargs)
 
-        return r.iter_lines()
+        lines = r.iter_lines()
+        return (l.decode('utf8') if isinstance(l, six.binary_type)
+                else l for l in lines)
```

Comment:
I found it more appropriate to handle them internally as unicode strings. Also, it looks more convenient in certain cases.

For example, in `serialization.py` [1], `json.loads()` expects a unicode string in Python 3+:
```

In [4]: json.loads(b'{"a":"a"}')
(...)
raise JSONDecodeError("Unexpected UTF-8 BOM (decode using utf-8-sig)",
TypeError: the JSON object must be str, not 'bytes'
```
On the other hand, in Python 2 `json.loads()` accepts both (unicode and bytes), but always generates objects with unicode strings:

```
In [4]: json.loads(b'{"a":"a"}')
Out[4]: {u'a': u'a'}

In [5]: json.loads(u'{"a":"a"}')
Out[5]: {u'a': u'a'}
```

[1] https://github.com/scrapinghub/python-hubstorage/blob/6098e59db1ad9bd04a60b650c4797d2cd7437f10/hubstorage/serialization.py#L17